torch_xla/csrc/pooling.cpp

```diff
@@ -538,59 +540,92 @@ xla::XlaOp BuildAvgPoolNdBackward(xla::XlaOp out_backprop, xla::XlaOp input,
                             /*spatial_dim_count=*/spatial_dim_count);
 }
 
-xla::XlaOp BuildAdaptiveAvgPool2d(xla::XlaOp input,
-                                  absl::Span<const xla::int64> output_size) {
-  XLA_CHECK_EQ(output_size.size(), 2) << "Invalid output size rank";
+xla::XlaOp BuildAdaptiveAvgPool(xla::XlaOp input,
+                                absl::Span<const xla::int64> output_size,
+                                int unsigned pool_dim) {
   const auto input_size = XlaHelpers::SizesOfXlaOp(input);
```

Comment:
This can be optimized by using `XlaHelpers::ShapeOfXlaOp()` and then use `rank()` and `dimensions()` instead of creating a vector copy of the dimensions.

Reply:
Forgive me for a naive question: Wouldn't we be creating copy of something more. XlaHelpers::ShapeOfXlaOp() returns an object with rank() and dimension() methods. Dimension method return the vector that we are using here. If create a copy of shape object instead of this vector, what did we save?

Reply:
I think you mean creating a reference to shape once and using it everywhere else?

Reply:
But in that case we can use the reference to the dimension vector itself?

Reply:
That API returns a const reference to shape data stored within the XLA builder. There won't be copies.

Reply:
ok. This makes sense. I will make this update.